Sources/ODP/ZaiusGraphQLApiManager.swift

```diff
@@ -0,0 +1,238 @@
+        
+        let session = self.getSession()
+        // without this the URLSession will leak, see docs on URLSession and https://stackoverflow.com/questions/67318867
+        defer { session.finishTasksAndInvalidate() }
+
+        let task = session.dataTask(with: urlRequest) { data, response, error in
+            var returnError: OptimizelyError?
+            var returnSegments: [String]?
+            
+            defer {
+                completionHandler(returnSegments, returnError)
+            }
+
+            guard error == nil, let data = data, let response = response as? HTTPURLResponse else {
+                let msg = error?.localizedDescription ?? "invalid response"
+                self.logger.d {
+                    "GraphQL download failed: \(msg)"
+                }
+                returnError = .fetchSegmentsFailed("network error")
+                return
+            }
+            
+            let status = response.statusCode
+            guard status < 400 else {
+                returnError = .fetchSegmentsFailed("\(status)")
+                return
+            }
+            
+            guard let dict = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] else {
+                returnError = .fetchSegmentsFailed("decode error")
+                return
+            }
+            
+            // most meaningful ODP errors are returned in 200 success JSON under {"errors": ...}
+            if let odpErrors: [[String: Any]] = dict.extractComponent(keyPath: "errors") {
+                if let odpError = odpErrors.first, let errorClass: String = odpError.extractComponent(keyPath: "extensions.classification") {
+                    if errorClass == "InvalidIdentifierException" {
+                        returnError = .invalidSegmentIdentifier
+                    } else {
+                        returnError = .fetchSegmentsFailed(errorClass)
+                    }
+                    return
+                }
+            }
+            
+            guard let audDict: [[String: Any]] = dict.extractComponent(keyPath: "data.customer.audiences.edges") else {
+                returnError = .fetchSegmentsFailed("decode error")
+                return
+            }
+                    
+            let audiences = audDict.compactMap { OdpAudience($0["node"] as? [String: Any]) }
+            returnSegments = audiences.filter { $0.isQualified }.map { $0.name }
+        }
+        
+        task.resume()
+    }
+    
+    func getSession() -> URLSession {
+        return URLSession(configuration: .ephemeral)
+    }
+    
+    func makeSubsetFilter(segments: [String]) -> String {
+        // segments = []: (fetch none)
+        //   --> subsetFilter = "(subset:[])"
+        // segments = ["a"]: (fetch one segment)
+        //   --> subsetFilter = "(subset:[\"a\"])"
+
+        let serial = segments.map { "\"\($0)\""}.joined(separator: ",")
```

Comment:
I think here we need to create a space behind the comma in the separator: .joined(separator: ", ").
The example response has space after the comma: 
https://github.com/optimizely/swift-sdk/pull/455/files#diff-7b392d5a916e5d85309b1f1a6f2dafa7581f486ddb872476725a11acf8dc7367R35

 don't see you stripping empty space. Is space behind comma important?

Reply:
The spaces in the samples are for readability only. No need for space. The samples fixed to avoid confusion.